src/server/components/dashboard/controllers/__tests__/listsControllers.helpers.test.ts

```diff
@@ -0,0 +1,114 @@
+import { calculateSortOrder, convertEntryToObject, sanitiseQuery, tableHeaders } from "../listsController.helpers";
+
+describe("calculateSortOrder", () => {
+  test("returns correct sort order when there are no query parameters", () => {
+    const expectedResult = [{ country: "asc" }, { type: "asc" }];
+    expect(calculateSortOrder({})).toEqual(expectedResult);
+  });
+
+  test("ignores unaccepted query parameters", () => {
+    const expectedResult = [{ country: "asc" }, { type: "asc" }];
+    expect(calculateSortOrder({ jsonData: "asc" })).toEqual(expectedResult);
+  });
+
+  test("returns correct sort order with desc sort direction", () => {
+    const expectedResult = [{ admins: "desc" }, { country: "asc" }, { type: "asc" }];
+    expect(calculateSortOrder({ admins: "desc" })).toEqual(expectedResult);
+  });
+
+  test("returns correct sort order with asc sort direction", () => {
+    const expectedResult = [{ admins: "asc" }, { country: "asc" }, { type: "asc" }];
+    expect(calculateSortOrder({ admins: "asc" })).toEqual(expectedResult);
+  });
+
+  test("unknown values are stripped", () => {
+    expect(calculateSortOrder({ admins: "abcef" })).toEqual([{ country: "asc" }, { type: "asc" }]);
+    expect(calculateSortOrder({ country: "abcef" })).toEqual([{ country: "asc" }, { type: "asc" }]);
+    expect(calculateSortOrder({ type: "abcef" })).toEqual([{ country: "asc" }, { type: "asc" }]);
```

Comment:
ensuring that even if country or type would've been "overwritten" - the value is stripped and does not affect the default sort order.